### ISSUE TITLE
GCC 7 fails to install on 10.6 Intel at stage3

### ISSUE THREAD

**Issue by mistydemeo** [mistydemeo/tigerbrew#554]
```
warning: gcc/cc1obj-checksum.o differs
warning: gcc/cc1objplus-checksum.o differs
Bootstrap comparison failure!
gcc/cp/semantics.o differs
make[2]: *** [compare] Error 1
make[1]: *** [stage3-bubble] Error 2
make: *** [all] Error 2
```

Logs: https://gist.github.com/a39f7818b6290e7f2a98124a9eb3acfb

GCC 5 and 6 both install fine.

**Comment by aeggenberger** [comment]
I'm getting this one too. Same system and error.

**Comment by sevan** [comment]
Yep, just hit this. Tried to turn on dwarf2 support & it didn't help.

**Comment by sevan** [comment]
I suspect the issue is with one of the configure options in the formula. I tried building the formula without patches, didn't help.
I build GCC manually, outside of tigerbrew, but still used its dependencies from (such as gmp, mpfr) and it managed to build and install.

```
Using built-in specs.
COLLECT_GCC=Sandbox/stock-gcc7/bin/gcc-7
COLLECT_LTO_WRAPPER=/Users/sme/Sandbox/stock-gcc7/libexec/gcc/x86_64-apple-darwin10.8.0/7.5.0/lto-wrapper
Target: x86_64-apple-darwin10.8.0
Configured with: ../configure --prefix=/Users/sme/Sandbox/stock-gcc7 --program-suffix=-7 --with-gmp=/Users/sme/tigerbrew/opt/gmp --with-mpfr=/Users/sme/tigerbrew/opt/mpfr --with-mpc=/Users/sme/tigerbrew/opt/libmpc --with-isl=/Users/sme/tigerbrew/opt/isl --with-system-zlib
Thread model: posix
gcc version 7.5.0 (GCC) 
```

**Comment by sevan** [comment]
> I had a bunch of other experiments in and have cleaned up my tree with just these changes in place for one last build to confirm. Will know in about 3 hours if anything else that I took out was needed. :)

It worked. Just need to turn off compiler optimisation in `gcc.rb` and `Library/Homebrew/extend/ENV/super.rb`

**Comment by sevan** [comment]
I suspect we might be able to work around this by setting `ENV.O0`. It would be nice if the default optimisation settings GCC itself expects could be used on Snow Leopard.

**Comment by sevan** [comment]
Previous comment is wrong, the issue with `ENV.O0` is that it applies `-O0` for everything, where as GCC build itself only applies optimisation on certain files.
I had a go at porting `no_optimization` to `superenv` this morning, where if `no_optimization` is set, `$HOMEBREW_OPTIMIZATION_LEVEL` is deleted from the build env, and the `cc` wrapper skips setting an optimisation level.
There's a build of GCC 7 in progress on Snow Leopard and so far it looks like it's working, according to `02.make.cc`.

**Comment by sevan** [comment]
It worked, GCC 7 built with `ENV.no_optimization` set & PR https://github.com/mistydemeo/tigerbrew/pull/1546 on Snow Leopard

**Comment by sevan** [comment]
As per previous comment, GCC 7 built fine out of the box with the proposed change.
